Add 'experimental' note to extensions chapter (#11322)
bokeh/bokeh: sphinx/source/docs/user_guide/extensions.rst

* Create highly specialized models for domain-specific use cases

You can make and use custom extensions with standard releases and don't need to
set up a development environment or build anything from source. This is the

> easiest way to get involved in Bokeh development. You can try new features and

Author: FYI: I removed "This is the easiest way" because this would now contradict the new note ("is a highly advanced feature").

Reviewer: I'm not sure they are contradictory. They are both an advanced feature for regular users, at the same time as being a good way for aspiring BokehJS devs to get started (since there is no need to set up a dev env)

having to wait for the core team to implement them into Bokeh itself.

.. note::
    Extending Bokeh is a highly advanced feature. Several aspects of creating
    and using extensions are still under development and should be considered

>     experimental.

Reviewer: Is it worth enumerating things we *do* consider "stable"?

* existing models and properties
* `change.emit`, etc.

Author: Do you mean a list of stable elements of Bokeh in general or of stable elements in the context of developing and using extensions? 

I think a general list of what we consider stable/testing/experimental/etc in Bokeh would almost be too broad for the extensions chapter. If we want to start having a definitive list of which features we consider stable or experimental, something like the release notes, "Defining Key Concepts" in the user guide, and/or the developers guide might be a better place for that.

If you are talking about a list of stable elements to rely on specifically when developing and using extensions, I'd appreciate some more information on what to include in that list.

Reviewer: I mean a list of things that are relatively safe to do in extensions. I've suggested a couple of things above, but we would need some input from @mattpap to make confident claims.

Reviewer 2: We may want to make this a 3.0 task, and formally define the public facing APIs (beyond the trivia like modes and properties).

Reviewer: OK I am fine to wait. My only other comment @tcmetzger is a few wording suggestions

>     Extending Bokeh is a highly advanced feature. Several aspects of creating
>     and using extensions are still under development and should be considered
>     experimental.

Reviewer: ```suggestion
    Extending Bokeh is an advanced feature. Some aspects of creating and using
    extensions are still under active development and should be considered
    experimental.
```
